src/memory.rs

```diff
@@ -238,6 +234,23 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
         }))
     }
 
+    pub fn create_fn_ptr_from_noncapture_closure(&mut self, tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId, substs: ClosureSubsts<'tcx>, fn_ty: ClosureTy<'tcx>) -> Pointer {
+        // FIXME: this is a hack
+        let fn_ty = tcx.mk_bare_fn(ty::BareFnTy {
+            unsafety: fn_ty.unsafety,
+            abi: fn_ty.abi,
+            sig: fn_ty.sig,
+        });
+        self.create_fn_alloc(Function::NonCaptureClosureAsFnPtr(FunctionDefinition {
+            def_id,
+            substs: substs.substs,
+            abi: Abi::Rust, // adjust abi
+            // FIXME: why doesn't this compile?
+            //sig: tcx.erase_late_bound_regions(&fn_ty.sig),
+            sig: fn_ty.sig.skip_binder(),
+        }))
```

Comment:
I'm not sure this is the way to go. Did you see how this was implemented in `librustc_trans`?
That is, it relies on an existing shim- oh, drat, you have that extra argument to take care of, nevermind me!